### DISCUSSION TITLE
How to exclude css modules from being bundled in js?

### DISCUSSION THREAD

**Discussion by akaSybe** [parcel-bundler/parcel#5246] [Q&A]
Is there some flag to disable bundling css-modules into js-bundle?

**Comment by aminya** [comment]
I have the same problem. I have a library that imports CSS into the components. I do this so Parcel transforms and bundles CSS for me. But I don't want them to be required into the JS bundle which breaks the workflow of anyone who does not use Parcel or similar bundlers.

https://github.com/aminya/solid-simple-table/blob/20c4c543fe4604acd3877426055abc474416d67a/src/SimpleTable.tsx#L2

**Reply by chisholmd** [reply]
I have a similar request. I have config.js files that I don't want bundled.

**Reply by aminya** [reply]
If it is a file you can use alias in package.json

```
alias: { "./src/config.js": false }
```

For css see this:
https://github.com/parcel-bundler/parcel/issues/5590#issuecomment-757084487

**Reply by chisholmd** [reply]
Thanks I will try that...  I am expecting this to maintain the import statement in the app, but for parcel to not bundle it.

**Reply by aminya** [reply]
@chisholmd No, it actually removes it.

**Reply by chisholmd** [reply]
Hmmm well when I need to I want to be able to tell a client that they can customize the app by going to the app directory, open the client_config.js file which will look like this:
`export default {} `
and then they can add custom settings, for example
```
export default {
      idle_timeout: 60
} 
```

What happens now is that the  client_config.js file is merged and bundled with the rest, I really need it to be a stand alone file. larger clients would then be able to ship config files to other installations etc.

Anyway to exclude a file so that they existing import statement is left original?

**Reply by aminya** [reply]
Not sure if I understand your problem. There are externals and aliases options that are supported by Parcel. 
https://v2.parceljs.org/features/module-resolution/

**Reply by chisholmd** [reply]
I will give that a try.   But excluding a file using "includeNodeModules" is a little unintuitive lol

**Reply by chisholmd** [reply]
I expect I am being thick headed about this but... if my source has an import statement like:
import * as clientConfig from '/config/client_config.js'

and I want that file to still be imported as a separate file after bundling...  
"alias": {
   "my-config-module": { "fileName": "./config/client_config.js" }
   }
Is that all I have to do?

**Reply by chisholmd** [reply]
I think maybe I need to use some sort of dynamic import instead

**Reply by chisholmd** [reply]
I realize that I am also leaving out a critical part of the requirement.   Currently I am using successive imports to override previous values in my initial state
```
initialState.js
        import * as defaults from '/config/defaults.js'
        import * as clientConfig from '/config/client_config.js'
        import * as userConfig from '/config/user_config.js'
        
        export default {
            ...defaults.default,
            ...clientConfig.default,
            ...userConfig.default
        }
```
Where \config\defaults.js might have:
`Idle_timeout: 90`

But \config\clientConfig.js might have:
`Idle_timeout: 30`

Because of the order of import, the client configs only need to create entries for values they want changed from the default.

And the file has to be unbundled so that local administrators can easily find and edit them.  So they need to be not bundled, and loaded in a specific order.

The moer I look at this the more dynamic import seems to make sense,

**Reply by chisholmd** [reply]
Damn.  I thought it was looking good to have a config loader run when the app init's to dynamically import modules....

```
  import('../config/user_config.js').then((userConfig) => {
            Object.keys(userConfig.default).forEach(function(key) {
                storeInstance.dispatch("setState", {property: key , value: userConfig.default[key] } )
            });
        })
```
...but no, it looks like parcel read this and bundled them anyway.
